### PR TITLE
adding all files for color sensor

### PR DESCRIPTION
**Pull request by shoryamalani**

### CODE REVIEW

**Review by WAaron-500 — CHANGES_REQUESTED** [review]
Don't use Integer its ugly use int

File: src/main/java/frc/robot/subsystems/colorSensor/ColorSensor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public ColorSensor(ColorSensorIO colorSensorIO){
+        this.colorSensorIO = colorSensorIO;
+
+
+    }
+    public Color getColor(){
+        // NOT FINISHED 
+        return colorSensorIO.getColor();
+    }
+    public Integer getProximity(){
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/colorSensor/ColorSensor.java]:
```suggestion
    public int getProximity(){
```

File: src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIO.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package frc.robot.subsystems.colorSensor;
+import edu.wpi.first.wpilibj.util.Color;
+public interface ColorSensorIO {
+    public default Color getColor() {
+        return Color.kBlack;
+    }
+    public default Integer getProximity() {
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIO.java]:
```suggestion
    public default int getProximity() {
```

File: src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIORevV3.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ColorSensorV3 m_colorSensor;
+    ColorMatch m_colorMatcher;
+    private final Color kBlueTarget = new Color(0.143, 0.427, 0.429);
+    private final Color kRedTarget = new Color(0.561, 0.232, 0.114);
+    public ColorSensorIORevV3(){
+        this.m_colorSensor = new ColorSensorV3(I2C.Port.kOnboard);
+        this.m_colorMatcher = new ColorMatch();
+
+    }
+    public Integer getProximity(){
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIORevV3.java]:
```suggestion
    public int getProximity(){
```

File: src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIORevV3.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.revrobotics.ColorMatchResult;
+import com.fasterxml.jackson.databind.deser.DataFormatReaders.Match;
+import com.revrobotics.ColorMatch;
+import frc.robot.Constants;
+
+public class ColorSensorIORevV3 implements ColorSensorIO {
+    ColorSensorV3 m_colorSensor;
+    ColorMatch m_colorMatcher;
+    private final Color kBlueTarget = new Color(0.143, 0.427, 0.429);
+    private final Color kRedTarget = new Color(0.561, 0.232, 0.114);
```
Comment (hadley31) [inline src/main/java/frc/robot/subsystems/colorSensor/ColorSensorIORevV3.java]:
```suggestion
    private static final Color kBlueTarget = new Color(0.143, 0.427, 0.429);
    private static final Color kRedTarget = new Color(0.561, 0.232, 0.114);
```

**Review by hadley31 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**